### ISSUE TITLE
Failed to fallback to ws

### ISSUE THREAD

**Issue by nobody-65534** [XTLS/Xray-core#249]
Server config based on [this example](https://github.com/XTLS/Xray-examples/tree/main/VLESS-TCP-XTLS-WHATEVER):
```json
{
    "inbounds":[
        {
            "port":"443",
            "protocol":"vless",
            "settings":{
                "decryption":"none",
                "clients":[
                    {
                        "id":"my-u-u-i-d",
                        "flow": "xtls-rprx-direct"
                    }
                ],
                "fallbacks":[
                    {
                        "dest":"/dev/shm/nginx.sock"
                    },
                    {
                        "path":"/xray_ws",
                        "dest":"/dev/shm/xray_ws.sock",
                        "xver":1
                    }
                ]
            },
            "streamSettings":{
                "network":"tcp",
                "security":"xtls",
                "xtlsSettings":{
                    "alpn": [
                        "http/1.1"
                    ],
                    "certificates":[
                        {
                            "certificateFile":"/path/to/fullchain.crt",
                            "keyFile":"/path/to/key.pem"
                        }
                    ]
                }
            }
        },
        {
            "listen":"/dev/shm/xray_ws.sock",
            "protocol":"vless",
            "settings":{
                "decryption":"none",
                "clients":[
                    {
                        "id":"my-u-u-i-d"
                    }
                ]
            },
            "streamSettings":{
                "network":"ws",
                "security":"none",
                "wsSettings":{
                    "acceptProxyProtocol": true,
                    "path":"/xray_ws"
                }
            }
        }
    ],
    "outbounds":[
        {
            "protocol":"freedom"
        }
    ]
}
```

vless + xtls + tcp works fine, and it falls back to nginx w/o problem. 

However, I failed to use the websocket with the following config:

```
vless://my-u-u-i-d@mydomain.com:443?path=%2Fxray_ws&security=tls&encryption=none&host=mydomain.com&type=ws#CDN
```

Server log:

```
2021/02/13 02:08:16 [Info] [3034811838] proxy/vless/inbound: firstLen = 456                                                                                                                            
2021/02/13 02:08:16 [Info] [3034811838] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2021/02/13 02:08:16 [Info] [3034811838] proxy/vless/inbound: realName = mydomain.com
2021/02/13 02:08:16 [Info] [3034811838] proxy/vless/inbound: realAlpn =                                                                                                                                
2021/02/13 02:08:16 [Info] [3034811838] proxy/vless/inbound: realPath = /xray_ws
2021/02/13 02:08:17 [Info] [3034811838] app/proxyman/inbound: connection ends > proxy/vless/inbound: fallback ends > context canceled
```

Client log:

```
2021/02/13 01:19:47 [Warning] [1418920296] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://mydomain.com/xray_ws): 400 Bad Request > websocket: bad handshake transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://mydomain.com/xray_ws):  > dial tcp: operation was canceled] > common/retry: all retry attempts failed
2021/02/13 01:19:47 [Warning] [4234900703] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://mydomain.com/xray_ws): 400 Bad Request > websocket: bad handshake transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://mydomain.com/xray_ws):  > dial tcp: operation was canceled] > common/retry: all retry attempts failed
```

Server/Client version: `Xray 1.2.4 (Xray, Penetrates Everything.) Custom (go1.15.7 linux/amd64)`

CDN Provider: Cloudflare

**Comment by James537** [comment]
It is impossible to run tcp xtls through CDN directly.

**Comment by nobody-65534** [comment]
> It is impossible to run tcp xtls through CDN directly.

I understand that CDN is unable to handle xtls + tcp. I'm expecting the ws + tls traffic with path `/xray_ws` falls back to the second inbound object. I tried accessing https://mydomain.com/ through CDN and it works fine. Use browser to open https://mydomain.com/xray_ws returns `400 Bad Request` which is expected. BTW I still cannot make the fallback work even without the CDN.

**Comment by ljxfstorm** [comment]
Comparing to my working config and [the example config](https://github.com/XTLS/Xray-examples/blob/main/VLESS-TCP-XTLS-WHATEVER/config_server.json), could you add "xver" to the default nginx fallback object and take a try then?

**Comment by 2444989513** [comment]
```
{
    "inbounds":[



        {
            "port":"443",
            "protocol":"vless",
            "settings":{
                "decryption":"none",
                "clients":[
                    {
                        "id":"my-u-u-i-d",
                        "flow": "xtls-rprx-direct"
                    }
                ],
                "fallbacks":[
				
//                    {
//                        "dest":"/dev/shm/nginx.sock"
//                    },
                    {
                        "path":"/xray_ws",
                        "dest":"6666",
                        "xver":1
                    }
					
                ]
            },
            "streamSettings":{
                "network":"tcp",
                "security":"xtls",
                "xtlsSettings":{
                    "alpn": [
                        "http/1.1"
                    ],
                    "certificates":[
                        {
                            "certificateFile":"/path/to/fullchain.crt",
                            "keyFile":"/path/to/key.pem"
                        }
                    ]
                }
            }
        },



        {
            "port": 6666,
            "listen": "127.0.0.1",
		
//            "listen":"/dev/shm/xray_ws.sock",
            "protocol":"vless",
            "settings":{
//                "decryption":"none",
                "clients":[
                    {
                        "id":"my-u-u-i-d"
                    }
                ],
                "decryption": "none"
            },
            "streamSettings":{
                "network":"ws",
                "security":"none",
                "wsSettings":{
                    "acceptProxyProtocol": true,
                    "path":"/xray_ws"
                }
            }
        }
    ],



    "outbounds":[
        {
            "protocol":"freedom"
        }
    ]


}
```

**Comment by nobody-65534** [comment]
Okay, I figured this one out. I added `"xver": 1` to the default fallback to the nginx, then add `proxy_protocol` in the nginx config according to [NGINX Docs](https://docs.nginx.com/nginx/admin-guide/load-balancer/using-proxy-protocol/#configuring-nginx-to-accept-the-proxy-protocol), now everything works like a charm. Thanks @ljxfstorm. 

However, before close this issue, I still wish someone could explain on why the `xver` in the default fallback to nginx will have an effect on the fallback to ws.
